Automation/SQL/azureDeploy.ps1

```diff
@@ -0,0 +1,24 @@
+param(
+        [string] [Parameter(Mandatory=$true)] $location,
+        [string] [Parameter(Mandatory=$true)] $resourceGroupName,
+        [string] [Parameter(Mandatory=$true)] $serverName,
+        [string] [Parameter(Mandatory=$true)] $tenantId,          
+        [string] [Parameter(Mandatory=$true)] $applicationId,
+        [string] [Parameter(Mandatory=$true)] $secret,
+        [string] [Parameter(Mandatory=$true)] $subscriptionId
+)
+
+$pathToAzSqlTemplate = "./azureDeploy.json"
+   
+& "../login.ps1" $tenantId $applicationId $secret $subscriptionId
+
+New-AzResourceGroup -Name $resourceGroupName -Location "$location"
+
+Write-host "Creating primary server..."
+New-AzSqlServer -ResourceGroupName $resourceGroupName `
+  -ServerName $serverName `
+  -Location $location `
+  -SqlAdministratorCredentials $(New-Object -TypeName System.Management.Automation.PSCredential `
```

Comment:
I get this error:
`Creating primary server...
ConvertTo-SecureString: D:\VSCode\ad440-winter2021-thursday-repo\Automation\SQL\azureDeploy.ps1:22:63
Line |
  22 |  … tList $adminLogin, $(ConvertTo-SecureString -String $password -AsPlai …
     |                                                        ~~~~~~~~~
     | Cannot bind argument to parameter 'String' because it is null.

New-Object: D:\VSCode\ad440-winter2021-thursday-repo\Automation\SQL\azureDeploy.ps1:21:34
Line |
  21 |  … edentials $(New-Object -TypeName System.Management.Automation.PSCrede …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling ".ctor" with "2" argument(s): "Cannot process argument because the value of argument
     | "userName" is not valid. Change the value of the "userName" argument and run the operation again."

New-AzSqlServer: D:\VSCode\ad440-winter2021-thursday-repo\Automation\SQL\azureDeploy.ps1:21:32
Line |
  21 |  … Credentials $(New-Object -TypeName System.Management.Automation.PSCre …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot validate argument on parameter 'SqlAdministratorCredentials'. The argument is null. Provide a
     | valid value for the argument, and then try running the command again.


cmdlet New-AzResourceGroupDeployment at command pipeline position 1
Supply values for the following parameters:
(Type !? for Help.)
administratorLogin: t
administratorLoginPassword: *
New-AzResourceGroupDeployment: D:\VSCode\ad440-winter2021-thursday-repo\Automation\SQL\azureDeploy.ps1:24:1
Line |
  24 |  New-AzResourceGroupDeployment -ResourceGroupName $resourceGroupName - …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | 12:00:20 AM - Error: Code=InvalidTemplateDeployment; Message=The template deployment failed with error:
     | 'Authorization failed for template resource '78d8f9e7-8d5e-50d2-b65f-02e9c2d4e2af' of type
     | 'Microsoft.Authorization/roleAssignments'. The client '26a56533-5906-448b-9a46-3f1c5f9fa135' with object
     | id '26a56533-5906-448b-9a46-3f1c5f9fa135' does not have permission to perform action
     | 'Microsoft.Authorization/roleAssignments/write' at scope
     | '/subscriptions/9f4dcf43-aa06-457b-b975-f0216baef20d/resourceGroups/yaholl-test-rg/providers/Microsoft.Authorization/roleAssignments/78d8f9e7-8d5e-50d2-b65f-02e9c2d4e2af'.'.

New-AzResourceGroupDeployment: D:\VSCode\ad440-winter2021-thursday-repo\Automation\SQL\azureDeploy.ps1:24:1
Line |
  24 |  New-AzResourceGroupDeployment -ResourceGroupName $resourceGroupName - …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The deployment validation failed`

Reply:
Should be fixed.